cmd/ui/src/App.tsx

```diff
@@ -14,47 +14,92 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 
-import { Box } from '@mui/material';
+import { Box, CssBaseline, ThemeProvider } from '@mui/material';
+import { createTheme } from '@mui/material/styles';
 import makeStyles from '@mui/styles/makeStyles';
+import {
+    AppNotifications,
+    GenericErrorBoundaryFallback,
+    NotificationsProvider,
+    lightPalette,
+    darkPalette,
+    typography,
+    components,
+} from 'bh-shared-ui';
+import { createBrowserHistory } from 'history';
 import React, { useEffect } from 'react';
+import { ErrorBoundary } from 'react-error-boundary';
 import { useQueryClient } from 'react-query';
-import { useLocation } from 'react-router-dom';
-import { AppNotifications } from 'bh-shared-ui';
-import Notifier from 'src/components/Notifier';
+import { unstable_HistoryRouter as BrowserRouter, useLocation } from 'react-router-dom';
```

Comment:
Nevermind. I thought we got rid of it but it just moved from main.tsx to here.

So another question. Both App.tsx and main.tsx are sort of these massive wrapper components around the entire application. Why do we have both and what are the responsibilities of each?

Reply:
The reason this was updated from what it was like before is so that we can use hooks to pull the dark mode state from the store so that it can inform the theme what palette to use. The theme was created in main.tsx previously and we can't use hooks there.